pyccel/ast/numpyext.py

```diff
@@ -786,25 +786,44 @@ def __new__(cls, a, dtype=None, order='K', subok=True, shape=None):
 class NumpyNorm(PyccelInternalFunction):
     """ Represents call to numpy.norm"""
     _dtype = NativeReal()
-
-    def __init__(self, arg, dim=None):
-        if isinstance(dim, ValuedArgument):
-            dim = dim.value
-        if self.dim is not None:
-            sh = list(sh)
-            del sh[self.dim]
+    _shape = ()
+    _rank  = 0
```

Comment:
Why do you need these class properties? Class properties should be used for the shape when the shape is always the same. This is not the case here as you set the shape below